woocommerce/woocommerce-ios
# [Mobile Payments] Disconnect from reader

Closes #4031 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader and uploading a pre-release version of WCPay to a test site⚠️

Add support for disconnecting from the currently connected reader:

https://user-images.githubusercontent.com/2722505/117460522-99d7cf80-af1a-11eb-9052-a9db2eb8dba2.MP4

## Changes
* Implemented `disconnect()` in the integration with the Stripe Terminal SDK (StripeCardReaderService)
* Added an action to CardPresentPaymentAction and its implementation to CardPresentPaymentStore.
* No new tests. Testing the integration is still blocked pending #3864 
* Touched the current UI to integrate with the new action. More about that in a comment in the actual diff.

## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. The pre-release version of WCPay that implements the endpoint that captures the payment id (see p91TBi-54R#comment-4441-p2 for a downloadable zip)
3. An external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Tap the "Disconnect" button. 
* The UI should flip to its initial state. The reader led should start blinking again. 
* Try to connect and disconnect a couple more times, just for kicks and giggles.

Update release notes:
